SIPS/sip-59.md

```diff
@@ -0,0 +1,141 @@
+
+## Simple Summary
+
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the SIP.-->
+
+Make upgrades of Synthetix as seamless as possible
+
+## Abstract
+
+<!--A short (~200 word) description of the technical issue being addressed.-->
+
+The introduction of the Delegated Migrator contract to perform upgrades on-chain in one or more steps. Migration scripts, once proposed and after a waiting period has elapsed, can be executed by the ProtocolDAO which will delegate to the migration scdript.
+
+## Motivation
+
+<!--The motivation is critical for SIPs that want to change Synthetix. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SIP solves. SIP submissions without sufficient motivation may be rejected outright.-->
+
+Upgrading the Synthetix protocol currently takes the core contributors a number of hours to run.
+
+A typical upgrade involves replacing the following contract sources: `Synthetix`, `FeePool` and every synth's `Synth` (over 40 at the time of writing), each of which require a change to both their associated proxies (via `setTarget`) and their associated state contracts (via `setAssociatedContract`). There are also changes to populate the `AddressResolver` with the new addresses, and updating each contract that depends on the AddressResolver to sync its cache (thus saving gas for end users).
+
+All of the above are tasks performed by the ProtocolDAO, which being a mulitisig safe needing signing by hardware keys of sufficient members, requires manual processing.
+
+Having a contract perform these upgrades is advantageous because not only does it lead to less downtime and has rollbacks built in, it also paves the way towards atomic upgrades, as future improvements to the protocol require less owner actions.
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature.-->
+
+In principle the idea is as follows:
+
+1. A contract is created, called the `DelegatedMigrator`, which becomes the owner of all Synthetix contracts. It implements `IDelegatedMigrator` (interface below)
+2. The owner of this contract is the [ProtocolDAO](https://etherscan.io/address/protocoldao.snx.eth)
+3. This migrator then allows the submission of proposals, in the form of a contract address. Each proposal must conform to `IMigration` (see below).
+4. After a waiting period has expired for a proposal, `execute` may be invoked by the ProtocolDAO sequentially for each step in the migration.
```

Comment:
If ProtocolDAO disagrees with a proposal that is voted on and passes, does it give the pDAO veto powers if it chooses to not execute each step in the migration?

Reply:
So in this iteration of the delegated migrator the proposals come from the pDAO and the SNX stakers can veto it in the waiting period.